File: src/test/test_battlecode.py
from battlecode import Piece, TypePiece, fight

def test_fight():
    p1 = Piece(TypePiece.SORCERESS)
    p2 = Piece(TypePiece.KNIGHT)
    assert fight(p1,p2) == p1
    assert fight(p2,p1) == p1
    p1 = Piece(TypePiece.DRAGON)
    p2 = Piece(TypePiece.TRAP)
    assert fight(p1,p2) == p1
    p1 = Piece(TypePiece.DWARF)
    assert fight(p1,p2) == p2
    p2 = Piece(TypePiece.DWARF)
    assert fight(p1,p2) == None
    assert fight(p2,p1) == None
    p2 = Piece(TypePiece.TREASURE)
    assert fight(p1,p2) == p2

    p23 = Piece(TypePiece.DRAGON)
    p24 = Piece(TypePiece.SLAYER)
    assert fight(p24,p23) == p23

File: test_battlecode.js
describe("the battlecode fight function", function() {
  it("decides who wins", function() {
    p1 = new Piece(TypePiece.SORCERESS)
    p2 = new Piece(TypePiece.KNIGHT)
    expect(fight(p1,p2)).toBe(p1)
    expect(fight(p2,p1)).toBe(p1)
    p1 = new Piece(TypePiece.DRAGON)
    p2 = new Piece(TypePiece.TRAP)
    expect(fight(p1,p2)).toBe(p1)
    p1 = new Piece(TypePiece.DWARF)
    expect(fight(p1,p2)).toBe(p2)
    p2 = new Piece(TypePiece.DWARF)
    expect(fight(p1,p2)).toBeUndefined()
    expect(fight(p2,p1)).toBeUndefined()
    p2 = new Piece(TypePiece.TREASURE)
    expect(fight(p1,p2)).toBe(p2)
    p23 = new Piece(TypePiece.DRAGON)
    p24 = new Piece(TypePiece.SLAYER)
    expect(fight(p24,p23)).toBe(p23)
  });
});
